Assets/HoloToolkit-Examples/GazeRuler/Scripts/LineManager.cs

```diff
@@ -20,6 +20,10 @@ public class LineManager : Singleton<LineManager>, IGeometry
 
         private const float defaultLineScale = 0.005f;
 
+        protected override void InitializeInternal()
```

Comment:
I'm not a big fan that we have this empty override method in each of the classes.

Is there a way to have the option of using this, without having to have an empty stub in every class that inherits from `Singleton`?

Reply:
We could, but InitializeInternal is meant to solve a problem with script access ordering, which it will only do if it is used to initialize the object.  Having it be optional to include would render it useless, as certainly nobody would use it.

I agree that it would be better to have had this from the start so they are used, and not empty.

Reply:
Idk, the script runtime execution order issues seem to be a one off for most classes, not all.

Option is better in this case, cause it's not something people will have to use every time.

Reply:
Simply making the abstract method virtual in the base Singleton Class should be enough.

Reply:
>Having it be optional to include would render it useless, as certainly nobody would use it.

I disagree, if there's a need, it'll get used.  We just need to make sure we document it correctly.